chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +193,250 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
 
         for block in more_blocks:
             await self.full_node.respond_block(RespondBlock(block))
+
+    async def process_blocks(
+        self, count: int, farm_to: bytes32 = bytes32([0] * 32), guarantee_transaction_blocks: bool = False
+    ) -> int:
+        """Process the requested number of blocks including farming to the passed puzzle
+        hash. Note that the rewards for the last block will not have been processed.
+        Consider `.farm_blocks()` or `.farm_rewards()` if the goal is to receive XCH at
+        an address.
+
+        Arguments:
+            count: The number of blocks to process.
+            farm_to: The puzzle hash to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos for the processed blocks.
+        """
+        rewards = 0
+        height = uint32(0)
+
+        if count == 0:
+            return rewards
+
+        for _ in range(count):
+            block: FullBlock
+            if guarantee_transaction_blocks:
+                block = await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
+            else:
+                block = await self.farm_new_block(FarmNewBlockProtocol(farm_to))
+            height = uint32(block.height)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+        peak_height = self.full_node.blockchain.get_peak_height()
+
+        if peak_height is None:
+            raise RuntimeError("Peak height still None after processing at least one block")
+
+        if peak_height < height:
+            raise RuntimeError(f"Peak height lower than expected: {peak_height} < {height}")
+
+        return rewards
+
+    async def farm_blocks_to_wallet(self, count: int, wallet: Wallet) -> int:
+        """Farm the requested number of blocks to the passed wallet. This will
+        process additional blocks as needed to process the reward transactions
+        and also wait for the rewards to be present in the wallet.
+
+        Arguments:
+            count: The number of blocks to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested address.
+        """
+        if count == 0:
+            return 0
+
+        target_puzzlehash = await wallet.get_new_puzzlehash()
+        rewards = 0
+
+        block_reward_coins = set()
+        expected_reward_coin_count = 2 * count
+
+        # TODO: why 2 and not 1?
+        for index in range(count + 2):
```

Comment:
It's 2 because each transaction will include all rewards up to and including the previous transaction block,

Reply:
`for index in range(count):` would iterate `count` times with all cases satisfying `index < count` and thus farming a block (maybe transaction, maybe not) to the wallet.  With `+ 1` it would iterate an extra time which should result in a final transaction block being processed (without the reward going to the wallt) that should include all the rewards from the previous `count` blocks.  `+ 2` processes a second transaction block.  Why do we need two transaction blocks at the end?

Though, I should probably write this a bit differently.  Reworded as below in https://github.com/Chia-Network/chia-blockchain/commit/980418d48de08c174c2831f1eb37ceb568475915..

```python
        # TODO: why two final transaction blocks and not just one?
        for to_wallet in [*([True] * count), False, False]:
            if to_wallet:
```